atlasdb-commons/src/main/java/com/palantir/common/compression/CompressorForwardingInputStream.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.common.compression;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+public class CompressorForwardingInputStream extends InputStream {
```

Comment:
this isn't the way to implement InputStream - basically you only implemented single byte read and so performance will end up an issue. Make sure you implemented the `read(buf, off, len)` method as well.

Reply:
done